react/useVideoTrack.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { RefObject, useState, useEffect } from 'react'
+
+interface State {
+  currentTime?: number
+  duration?: number
+}
+
+export const useVideoTrack = (
+  videoRef: RefObject<HTMLVideoElement>,
+  trackBarRef: RefObject<HTMLInputElement>
+) => {
+  const [state, setState] = useState<State>()
+  const currentVideoRef = videoRef?.current as any
+  const currentTrackBarRef = trackBarRef?.current as any
+
+  const updateMetaData = () => {
+    setState({ currentTime: 0, duration: currentVideoRef?.duration })
+  }
+
+  const updateCurrentTime = () => {
+    if (currentTrackBarRef) {
+      currentTrackBarRef.value = `${currentVideoRef?.currentTime}`
+    }
+
+    setState({ ...state, currentTime: currentVideoRef?.currentTime })
+  }
+
+  const setCurrentTime = (timeInSeconds: number) => {
+    if (!currentVideoRef && currentVideoRef !== 0) return
+
+    currentVideoRef.currentTime = timeInSeconds
+  }
+
+  const formatTime = (timeInSeconds: number) => {
+    const seconds: string = Math.floor(timeInSeconds % 60)
+      .toString()
+      .padStart(2, '0')
```

Comment:
Can you test this on IE11 to check if we have a polyfill? Or maybe do it any other way
https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Reply:
Not working =/ I will implement of another way, tks!